custom/icds/translations/integrations/transifex.py

```diff
@@ -52,11 +52,12 @@ def send_translation_files(self):
         """
         submit translation files to transifex
         """
-        app_trans_generator = AppTranslationsGenerator(
+        self.app_trans_generator = AppTranslationsGenerator(
             self.domain, self.app_id, self.version,
             self.key_lang, self.source_lang, self.lang_prefix,
             self.exclude_if_default, self.use_version_postfix)
-        with PoFileGenerator(app_trans_generator.translations, app_trans_generator.metadata) as po_file_generator:
+        with PoFileGenerator(self.app_trans_generator.translations,
+                             self.app_trans_generator.metadata) as po_file_generator:
```

Comment:
@millerdev @esoergel 
Considering the great conversations we have had last week,
Curious if you think its fine to 
just put `app_trans_generator` on `self` so that its available later or 
pass `app_trans_generator` in the argument and then use `app_trans_generator. slug_to_name` or
pass `app_trans_generator.slug_to_name` as the argument itself for use later?

Reply:
If you assign it to `self` then it becomes an order-dependent resource and will make maintaining this class more difficult because the life-cycle of that attribute is not immediately clear. My recommendation would be to pass it to the method(s) that need it rather than assign it to `self`.

Reply:
As you suggested, if you do need to keep around part of a class, I'd store just the simple attributes you need, such as in this case `slug_to_name`.  That still runs into the issue @millerdev mentioned - pylint gives a warning for class attributes defined outside of `__init__` for that reason, actually.